### ISSUE TITLE
Documenting an object parameter with the name `module` triggers `jsdoc/valid-types`

### ISSUE THREAD

**Issue by edg2s** [gajus/eslint-plugin-jsdoc#806]
## Expected behavior
No error is found in the code example

## Actual behavior
An error is reported from the line containing ` * @param {string} options.module`:
```
Syntax error in namepath: options.module
```

## ESLint Config

```js
{
  "jsdoc/valid-types": "error"
}
```

## ESLint sample

```js
/**
 * @param {string} id
 * @param {Object} options
 * @param {boolean} options.isSet
 * @param {string} options.module
 */
function ( id, options ) {
}
```

## Environment

- Node version: 
- ESLint version 
- `eslint-plugin-jsdoc` version: 37.0.3

**Comment by brettz9** [comment]
I think this might be an issue for `jsdoc-type-pratt-parser` (for not allowing "module" in this position)... @simonseyock ?

As a workaround, you can use jsdoc's escaping mechanism:

```js
* @param {string} options."module"
```

**Comment by dog-s-life** [comment]
Same issue for `external`.

```javascript
/**
  * @param {boolean} requestBody.external blah blah.
  */
```

**Comment by c69-addepar** [comment]
i have filed the same issue for parser, it reproduces consistently:
https://github.com/jsdoc-type-pratt-parser/jsdoc-type-pratt-parser/issues/102

But not sure if the actual a bug is in parser, or in this plugin.

**Comment by brettz9** [comment]
Closing as tracking to be done in `jsdoc-type-pratt-parser`. (We regularly update, and if we don't just ping us.)
